uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/MsalTestApp.java

```diff
@@ -90,33 +90,75 @@ public String acquireToken(@NonNull final String username,
         scrollToElement(acquireTokenSilentButton);
+        Thread.sleep(2000);
         acquireTokenSilentButton.click();
         final UiObject result = UiAutomatorUtils.obtainUiObjectWithResourceId("com.msft.identity.client.sample.local:id/txt_result");
         return result.getText();
     }
 
     // click on button getUsers
-    public List<String> getUsers() throws UiObjectNotFoundException {
-        UiAutomatorUtils.handleButtonClick("com.msft.identity.client.sample.local:id/btn_getUsers");
-
+    public List<String> getUsers() throws UiObjectNotFoundException, InterruptedException {
+        final UiObject getUsersButton = UiAutomatorUtils.obtainUiObjectWithResourceId("com.msft.identity.client.sample.local:id/btn_getUsers");
+        scrollToElement(getUsersButton);
+        Thread.sleep(2000);
+        getUsersButton.click();
         // get each user information in the user list
         final List<String> users = new ArrayList<>();
         final UiObject userList = UiAutomatorUtils.obtainUiObjectWithResourceId("com.msft.identity.client.sample.local:id/user_list");
+        Thread.sleep(2000);
         for (int i = 0; i < userList.getChildCount(); i++) {
             final UiObject user = userList.getChild(new UiSelector().index(i));
             users.add(user.getText());
         }
         return users;
     }
 
+    public void selectFromAuthScheme(String text) throws UiObjectNotFoundException {
+        final UiObject anthSchemeSpinner = UiAutomatorUtils.obtainUiObjectWithResourceId("com.msft.identity.client.sample.local:id/authentication_scheme");
+        anthSchemeSpinner.click();
+        final UiObject authScheme = UiAutomatorUtils.obtainUiObjectWithText(text);
+        authScheme.click();
+    }
+
+    public String generateSHR() throws UiObjectNotFoundException, InterruptedException {
+        final UiObject generateSHRButton = UiAutomatorUtils.obtainUiObjectWithResourceId("com.msft.identity.client.sample.local:id/btn_generate_shr");
+        scrollToElement(generateSHRButton);
+        Thread.sleep(2000);
+        generateSHRButton.click();
+        final UiObject result = UiAutomatorUtils.obtainUiObjectWithResourceId("com.msft.identity.client.sample.local:id/txt_result");
+        return result.getText();
+    }
+
+    public UiObject removeUser(String msg) throws UiObjectNotFoundException, InterruptedException {
+        final UiObject removeUserButton = UiAutomatorUtils.obtainUiObjectWithResourceId("com.msft.identity.client.sample.local:id/btn_clearCache");
+        scrollToElement(removeUserButton);
+        Thread.sleep(2000);
+        removeUserButton.click();
+        final UiObject toast = UiAutomatorUtils.obtainUiObjectWithText(msg);
+        return toast;
+    }
+
+    public UiObject getPackageName(String packageName) throws UiObjectNotFoundException, InterruptedException {
```

Comment:
what is the value passed for packageName? Is this just clicking on the getPackageName button? Why does it need an input parameter?

Reply:
I pass package name that expected(for example LTW's package name) so when click the button I can use it to obtain toast UiObject by calling `obtainUiObjectWithText`. If the UiObject exist means we got the right package name. In the test case it's used like this:
`        final UiObject activeBroker = msalTestApp.getPackageName(BrokerLTW.BROKER_LTW_APP_PACKAGE_NAME);
        Assert.assertTrue(activeBroker.exists());`

Reply:
can we modify this to simply click on the package name button and then return a string - which would be the value of the package name returned. And in your testcase, you can verify if the expected pkgName == retrieved pkgName